federation/pkg/federation-controller/sync/controller.go

```diff
@@ -356,10 +356,34 @@ func (s *FederationSyncController) reconcile(namespacedName types.NamespacedName
 		return statusError
 	}
 
+	schedObjsAccessor := func(adapter federatedtypes.FederatedTypeAdapter, fedObj pkgruntime.Object,
```

Comment:
This whole thing could better be executed inline here (or a function call), rather then being executed inside '''operationsAccessor()'''. '''operationsAccessor()''' can get the original set and selected set inplace to get the operations, which makes more sense going by the name of '''operationsAccessor''' (only gets operations and does not execute any object updates, into the federation or underlying clusters). [ to be taken care post replicaset/deployment pr merge ]